packages/flutter_tools/lib/src/build_system/targets/common.dart

```diff
@@ -222,7 +222,7 @@ class KernelSnapshot extends Target {
       ),
       aot: buildMode.isPrecompiled,
       buildMode: buildMode,
-      trackWidgetCreation: trackWidgetCreation && buildMode == BuildMode.debug,
+      trackWidgetCreation: trackWidgetCreation && buildMode != BuildMode.release,
```

Comment:
what performance impact does this have?

it changes the behaviour of constant widgets, which seems significant, for one

Reply:
Please do not land this until we've answered this question. Ideally, we'd know how much overhead _just_ this line adds, compared to the rest of this patch, as measured on a low end Android device with a large application like customer: money.

Reply:
> it changes the behaviour of constant widgets, which seems significant, for one

Good point.  As far as the Gallery is concerned, the change was negligible in local testing.  I'm open to suggestions to answering this question more definitively.  I can write a microbenchmark but what would be a reasonable way to exercise this?  A widget tree of 1000 const Widgets?

If this is a problem we can tweak track-widget-creation to store its data outside of the Widget.

Reply:
The change to const behavior would be much smaller if you stored a single bool with each widget indicating whether it is created from the local object or not. That way the only changes in const behavior would be cases where the same exact const widget instance was created by two packages one that happened to be local and one that didn't.

You can't track widget creation data outside the widgets and have it work for const widgets. You could track it outside the widgets for non-const widgets but the performance impact would be significantly higher due to the extra expando/hashmap lookup.